sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ProducerRecordCoder.java

```diff
@@ -58,7 +58,7 @@ public ProducerRecordCoder(Coder<K> keyCoder, Coder<V> valueCoder) {
   @Override
   public void encode(ProducerRecord<K, V> value, OutputStream outStream) throws IOException {
     stringCoder.encode(value.topic(), outStream);
-    intCoder.encode(value.partition() != null ? value.partition() : Integer.MAX_VALUE, outStream);
+    intCoder.encode(value.partition() != null ? value.partition() : -1, outStream);
```

Comment:
Your preference : either -1 or INT_MAX is fine with me.

Reply:
Let's keep "-1" for now